### ISSUE TITLE
feature request - current layer always visible and editable

### ISSUE THREAD

**Issue by pixsrv** [aseprite/aseprite#3575]
would be nice to have an option to set that current layer is always visible and editable

this should be an overriding feature set on header eye icon and lock icon
once set it should override visible / editable state of the selected (current) layer

the header icons should have this feature on Right Mouse Button click or on RMB popup (if there will be one)
and it should be marked on the header icons as different color or something even more distinctive

**Comment by Gasparoken** [comment]
Thank you for expressing your idea. So, to confirm my understanding:
This individual layer override will prevent the layer from being hidden/locked via the general Hide/Lock icon (at the top of the timeline).
I'm curious in what circumstances you should use this feature, in other words: how this feature will help you in your workflow.? (I ask because fully understanding your needs can lead to better solutions)

**Comment by pixsrv** [comment]
let's stick for a moment only to the locking layers

when user wants to protect his layers he locks them which is obvious and
great of course
but when user wants to switch to another layer then he needs to click twice
once to select the layer and once to unlock it

this second action can be omitted if the option "current layer always
visible and editable" would be switched on

now visibility (less common case)
when user selects layer (and its visibility is off) then selection has to
be accompanied by second click to switch visibility on
so the situation is identical to the locking

and the last and probably the most important thing
when user switches to another layer again then the old one becomes locked
automatically (because all are locked but only current is unlocked)
so again there is no need to click multiple times and the old layer is
protected again

this feature is not new - i use it all the time in my vector app (xara)
and it really do the job

**Comment by dacap** [comment]
A similar feature is in digital audio workstation (DAW) software to make the current layer the one that records. In some way this feature is related to the opacity level of non-active layers. E.g. the visibility option can be achieved right now putting this option to 0:

![image](https://user-images.githubusercontent.com/39654/199960121-a61ae5b8-d0ac-49c9-8e0c-6e1092113a79.png)

**Comment by pixsrv** [comment]
but visibility is not protecting
so as mentioned above - the visibility is less common
the most important and useful is the locking

and multiple clicking... to put one pixel on this layer and one on the other

**Comment by Gasparoken** [comment]
The lock feature was created to prevent the artist from accidentally working on the wrong layer.
So I think unlocking all layers has the same effect as 'current layer always editable'. 🤔 or... sorry, there is something I'm not understanding 😔.
Thinking about this feature, it can be useful if the artist wants to always work on one layer and wants to avoid accidentally transforming multiple layers/cells.

**Comment by pixsrv** [comment]
there's nothing to be sorry :)
people tends to resist the unknown and tends to stick to good-old stuff
i really get it
but in the last sentence sounds like you get my point

to be honest i work always on one layer at a time and i just want to be sure i really work on this particular one
without extra clicking looking or checking
i just always have them all locked and only the one i selected is unlocked

when it comes to the hidden layer
well... when user selects it... what is the purpouse of it if it's hidden... user has to click again to show it
so it does sound resonably to have it already visible, doesn't it? ;)

**Comment by Gasparoken** [comment]
Yes, I agree with you, the selected layer should be visible. This new behavior could be optional 👍.

**Comment by pixsrv** [comment]
that's great (even if imo visibility is less important that locking)
but... if you will implement 'current layer always visible' i hope you will also apply 'current layer always editable'
(fingers crossed)
;)
